crypto/evp/digest.c

```diff
@@ -227,6 +217,17 @@ static int evp_md_init_internal(EVP_MD_CTX *ctx, const EVP_MD *type,
 #endif
             || (ctx->flags & EVP_MD_CTX_FLAG_NO_INIT) != 0
             || type->origin == EVP_ORIG_METH) {
+        /* If we were using provided hash before, cleanup algctx */
```

Comment:
Just to confirm: if we follow the `skip_to_init` case ~20 lines up, that requires us to have been using an engine previously, which means we were not using a provided hash before?